### PR TITLE
Csr feat/fix serializer bug

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
Fix serialization bug for **KeysAndCertificate** operation introduced in [a recent PR](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/832)
Also, enable running unit and integration tests of this branch on TravisCI

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h [hunk 1]
```diff
@@ -66,9 +66,9 @@
     #define LIBRARY_LOG_LEVEL        AWS_IOT_LOG_LEVEL_PROVISIONING
 #else
     #ifdef IOT_LOG_LEVEL_GLOBAL
-        #define LIBRARY_LOG_LEVEL    IOT_LOG_LEVEL_GLOBAL
+        #define LIBRARY_LOG_LEVEL    IOT_LOG_DEBUG
```
Comment (aggarg) [inline libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h]: Why this change?

File: libraries/aws/provisioning/src/aws_iot_provisioning_serializer.c [hunk 2]
```diff
@@ -180,7 +180,7 @@ static AwsIotProvisioningError_t _serializeCreateKeysAndCertificateRequestPayloa
     /* Encode an empty map container (Diagnostic notation as "{}"") .*/
     serializerStatus = _pAwsIotProvisioningEncoder->openContainer( pOutermostEncoder,
                                                                    &emptyPayloadEncoder,
-                                                                   _numCertFromCsrMapEntries );
+                                                                   _numKeysAndCertMapEntries );
```
Comment (aggarg) [inline libraries/aws/provisioning/src/aws_iot_provisioning_serializer.c]: How was it compiling before?

Comment (aggarw13) [reply libraries/aws/provisioning/src/aws_iot_provisioning_serializer.c]: `_numCertFromCsrMapEntries` is a valid constant, but this specific function is related to the "KeysAndCertificate" operation.
I caught the error on my machine as I used the `-Wall` flag for building (This specifically was raised by the `-Wunused-const-variable=` flag

File: libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h (hunk 1 above)
Comment (aggarw13) [reply libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h]: Accidental. Will revert.

Comment (aggarw13) [reply libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h]: Reverted
